process/heartbeat/interceptedPeerAuthentication.go

```diff
@@ -0,0 +1,154 @@
+package heartbeat
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/hashing"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/heartbeat"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+// ArgInterceptedPeerAuthentication is the argument used in the intercepted peer authentication constructor
+type ArgInterceptedPeerAuthentication struct {
+	argBaseInterceptedHeartbeat
+}
+
+// interceptedPeerAuthentication is a wrapper over PeerAuthentication
+type interceptedPeerAuthentication struct {
+	peerAuthentication heartbeat.PeerAuthentication
+	peerId             core.PeerID
+	hash               []byte
+}
+
+// NewInterceptedPeerAuthentication tries to create a new intercepted peer authentication instance
+func NewInterceptedPeerAuthentication(arg ArgInterceptedPeerAuthentication) (*interceptedPeerAuthentication, error) {
+	err := checkBaseArg(arg.argBaseInterceptedHeartbeat)
+	if err != nil {
+		return nil, err
+	}
+
+	peerAuthentication, err := createPeerAuthentication(arg.Marshalizer, arg.DataBuff)
+	if err != nil {
+		return nil, err
+	}
+
+	intercepted := &interceptedPeerAuthentication{
+		peerAuthentication: *peerAuthentication,
+	}
+
+	intercepted.processFields(arg.Hasher, arg.DataBuff)
+
+	return intercepted, nil
+}
+
+func createPeerAuthentication(marshalizer marshal.Marshalizer, buff []byte) (*heartbeat.PeerAuthentication, error) {
+	peerAuthentication := &heartbeat.PeerAuthentication{}
+	err := marshalizer.Unmarshal(peerAuthentication, buff)
+	if err != nil {
+		return nil, err
+	}
+
+	return peerAuthentication, nil
+}
+
+func (ipa *interceptedPeerAuthentication) processFields(hasher hashing.Hasher, buff []byte) {
+	ipa.hash = hasher.Compute(string(buff))
+	ipa.peerId = core.PeerID(ipa.peerAuthentication.Pid)
+}
+
+// CheckValidity will check the validity of the received peer authentication. This call won't trigger the signature validation.
+func (ipa *interceptedPeerAuthentication) CheckValidity() error {
+	err := verifyPropertyLen(publicKeyProperty, ipa.peerAuthentication.Pubkey)
+	if err != nil {
+		return err
+	}
+	err = verifyPropertyLen(signatureProperty, ipa.peerAuthentication.Signature)
+	if err != nil {
+		return err
+	}
+	err = verifyPropertyLen(peerIdProperty, ipa.peerId.Bytes())
+	if err != nil {
+		return err
+	}
+	err = verifyPropertyLen(payloadProperty, ipa.peerAuthentication.Payload)
+	if err != nil {
+		return err
+	}
+	err = verifyPropertyLen(payloadSignatureProperty, ipa.peerAuthentication.PayloadSignature)
+	if err != nil {
+		return err
+	}
+
```

Comment:
we should also check the signatures here & if the signatures are worth of checking:
1. we should check if the BLS public key is a validator or not. This can be achieved by using an interface that defines the function `GetValidatorWithPublicKey(publicKey []byte) (validator Validator, shardId uint32, err error)` (it is implemented by the nodesCoordinator)
2. For the payload sig, we already have the implementation done in the netMessenger the possibility to check if the sig was done correctly, so one more interface should be passed on the argument, something that defines the 
`Verify(payload []byte, pid core.PeerID, signature []byte) error` method
Then, we need to check if the payload is a valid timestamp and that timestamp is still valid as it belongs to an interval [current time - expiry_duration, current time + small_threshold (might be a const of around 10 seconds)]
3. For the BLS sig, we should use the interface PeerSignatureHandler (it is already implemented by a component that is also used in the consensus mechanism). This also caches the bls sig, bls pk and the pid tuple. 
the BLS sig verification is deliberately left as the last check in order to check the sig only if the message is considered valid